packages/tooltip/src/vaadin-tooltip.js

```diff
@@ -60,13 +62,48 @@ class Tooltip extends ThemePropertyMixin(ElementMixin(PolymerElement)) {
         type: Object,
         observer: '__targetChanged',
       },
+
+      /**
+       * Set to true when the overlay is opened using auto-added
+       * event listeners: mouseenter and focusin (keyboard only).
+       * @protected
+       */
+      _autoOpened: {
+        type: Boolean,
+        observer: '__autoOpenedChanged',
+      },
     };
   }
 
   constructor() {
     super();
 
     this._uniqueId = `vaadin-tooltip-${generateUniqueId()}`;
+
+    this.__boundOnFocusin = this.__onFocusin.bind(this);
```

Comment:
nit: I guess it's a matter of preference, but the original function could just be overwritten: `this.__onFocusin = this.__onFocusin.bind(this);`
Makes it easier to discover later while debugging (no need to search for the original function name here).

Reply:
Thanks, changed.